src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -1,47 +1,161 @@
 package core.basesyntax;
 
 import java.util.List;
+import java.util.Objects;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> head;
+    private Node<T> tail;
+    private int size;
+
     @Override
     public void add(T value) {
+        Node<T> newNode = new Node<>(value);
+        if (head == null) {
+            head = newNode;
+        } else {
+            tail.next = newNode;
+            newNode.prev = tail;
+        }
+        tail = newNode;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        checkIndex(index);
+        if (index == size) {
+            add(value);
+        } else if (index == 0) {
+            Node<T> newNode = new Node<>(value, null, head);
+            head.prev = newNode;
+            newNode.next = head;
+            head = newNode;
+            size++;
+        } else {
+            Node<T> node = getNodeIndex(index);
+            Node<T> newNode = new Node<>(value, node.prev,node);
+            node.prev.next = newNode;
+            node.prev = newNode;
+            size++;
+        }
     }
 
     @Override
     public void addAll(List<T> list) {
+        for (T value: list) {
+            add(value);
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        checkIndexBound(index);
+        Node<T> node = getNodeIndex(index);
+        return node.value;
     }
 
     @Override
     public T set(T value, int index) {
-        return null;
+        checkIndexBound(index);
+        Node<T> currentNode = getNodeIndex(index);
+        T lastValue = currentNode.value;
+        currentNode.value = value;
+        return lastValue;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        checkIndexBound(index);
+        return unlinkNode(getNodeIndex(index));
     }
 
     @Override
     public boolean remove(T object) {
+        Node<T> currentNode = head;
+        for (int i = 0; i < size; i++) {
+            if (Objects.equals(object, currentNode.value)) {
```

Comment:
[checklist](https://mate-academy.github.io/jv-program-common-mistakes/java-core/collections/linked-list.html): Don’t use class Objects.

Reply:
What about this comment?